src/app/components/shell/friend-request-popover/friend-request-popover.ts

```diff
@@ -46,27 +51,68 @@ export default class AppShellFriendRequestPopover extends Vue {
 		return this.activeTab === 'requests' ? this.incoming : this.outgoing;
 	}
 
+	get isAtEnd() {
+		return (
+			(this.activeTab === 'requests' ? this.requestCount : this.pendingCount) ===
+			this.requests.length
+		);
+	}
+
 	async onShow() {
 		if (this.isLoading) {
 			return;
 		}
 
+		// Reset to empty.
+		this.incoming = [];
+		this.outgoing = [];
+
 		this.isShowing = true;
 		this.isLoading = true;
 
-		const { requests, pending } = await UserFriendship.fetchRequests();
-		this.incoming = requests;
-		this.setFriendRequestCount(this.incoming.length);
-		this.outgoing = pending;
+		// We only need to do this request once.
+		if (!this.isBootstrapped) {
+			const payload = await Api.sendRequest('/web/dash/friends/requests', null, {
+				detach: true,
+			});
+			this.requestCount = payload.requestCount;
+			this.pendingCount = payload.pendingCount;
+			this.perPage = payload.perPage;
+			this.isBootstrapped = true;
+
+			this.setFriendRequestCount(this.requestCount);
+		}
+
+		this.loadTab();
```

Comment:
Not awaiting on this means the popover is considered loaded before the first page actually loads. Is this intentional?

Reply:
No, good catch!